cypress/integration/pre-notice_spec.js

```diff
@@ -0,0 +1,99 @@
+// Pre notice tests
+
+describe('Pre notice tests', () => {
+    before(() => {
+        cy.login();
+    });
+
+    beforeEach(() => {
+        // Session cookies will not be cleared before the NEXT test starts
+        cy.preserveSessionOnce();
+    });
+
+
+    it('should render pre-notice list page', () => {
+        cy.visit('/#/pre-notices');
+        cy.contains('Omat säännöllisen henkilöliikenteen muutosilmoitukset');
+        cy.contains('Lähetetyt muutosilmoitukset');
+    });
+
+    it('should render pre-notice form page', () => {
+        cy.visit('/#/pre-notices');
+        cy.contains('Uusi muutosilmoitus').click();
+        cy.contains('Säännöllisen henkilöliikenteen muutosilmoitus');
+        cy.contains('Ilmoitettavan muutoksen tyyppi');
+        cy.contains('Muutoksen alkamispäivä tai -päivät');
+        cy.contains('Reitin ja alueen tiedot');
+        cy.contains('Muutoksen tarkemmat tiedot');
+        cy.contains('Tallenna ja lähetä');
+        cy.contains('Tallenna luonnoksena');
+
+    });
+
+
+    it('save and send pre-notice', () => {
+        // Open pre notice form
+        cy.visit('/#/pre-notices');
+        cy.contains('Uusi muutosilmoitus').click();
+
+        // Select type
+        cy.get('#0_termination').click();
+
+        // Select random effective date
+        cy.get('#effective-dates-table #row_0 input').first().click();
+        cy.document().its('body').type('{enter}');
+
+        // Give reason to make the change at given time
+        cy.get('#effective-dates-table #row_0 [name=effective-date-description]').type("Koska halusin");
+
+        // Describe route
+        cy.get('#route-description').type('Oulu - Kajaani');
+
+        // Select Route area from list
+        cy.get('[id*="Lismaakuntataimaakunnatjoitamuutoskoskee"]').type('Uusim{enter}');
+        cy.get('[id*="Lismaakuntataimaakunnatjoitamuutoskoskee"]').type('pohjanm{downarrow}{downarrow}{enter}');
+
+        cy.contains('Tallenna ja lähetä').click();
```

Comment:
Is this line needed? Duplicate to next line?

Reply:
That next line is confirmation modal. It has same text "Tallenna ja lähetä" so I need to identify it differently.